tutorials/polkadot-sdk/system-chains/asset-hub/batch-teleport-assets.md

```diff
@@ -0,0 +1,218 @@
+    "@paraspell/sdk": "^{{dependencies.javascript_packages.paraspell_sdk.version}}"
+  }
+}
+```
+
+## Configure Destination Addresses
+
+For testing purposes, you can also modify the destination addresses in the batch operations:
+
+- `'15whavTNSyceP8SL3Z1JukFcUPzmeR26RxKXkfQiPhsykg7s'`
+- `'14E5nqKAp3oAJcmzgZhUD2RcptBeUBScxKHgJKU4HPNcKVf3'`
+
+## Run Your Batch Teleport
+
+Execute the batch teleport from your project directory:
+
+```bash
+bun run dev
+```
+
+You should see output similar to:
+
+--8<-- 'code/tutorials/polkadot-sdk/system-chains/asset-hub/papi-console-output.html'
+
+The encoded data is formatted into a PAPI console link for easy submission through a web interface.
+
+Make sure your Polkadot wallet is funded with some test tokens by using the [Polkadot faucet](https://faucet.polkadot.io/?parachain=1000){target=\_blank}  and submit the transaction.
+
+## Submitting the Transaction
+
+After running the script:
+
+1. **Copy the generated link** from the console output
+2. **Open the link** in your web browser
+3. **Connect your wallet** (such as Polkadot.js extension)
+4. **Review the transaction details** in the PAPI console
+5. **Submit the transaction** and confirm with your wallet
+
+    ![](/images/tutorials/polkadot-sdk/system-chains/asset-hub/batch-teleport/papi-console-01.webp)
+
+The PAPI console will show:
+
+- The encoded call data
+- Transaction details and parameters
+- Estimated fees
+- Submission interface
+
+## Dry Run your Transactions
+
+It is best practice to conduct dry run tests before execution of a cross-chain transfer to prevent failures. XCM operations can trap or permanently lose assets if improperly constructed. The dry run functionality in this code helps identify potential issues before committing real assets to the cross-chain transfer.
+
+??? code "Batch Teleport with Dry Run"
+
+    ```typescript title="batch-teleport-with-dry-run.ts"
+    --8<-- 'code/tutorials/polkadot-sdk/system-chains/asset-hub/batch-teleport-with-dry-run.ts'
+    ```
+
+## Conclusion
+
+Congratulations! You've successfully implemented batch teleport functionality using the ParaSpell SDK. This single-file implementation demonstrates how to:
+
+- Bundle multiple cross-chain transfers into efficient batch operations
+- Configure network settings for different environments
+- Generate ready-to-submit transaction data
+- Use the PAPI console for transaction submission
+
+The batch teleport feature significantly improves efficiency when performing multiple cross-chain transfers, reducing both time and transaction costs in the Polkadot ecosystem.
+
+## Where to Go Next
+
+<div class="grid cards" markdown>
+
+-   <span class="badge guide">Guide</span> __Best Practices for Teleporting Assets__
+
+    ---
+
+    An in-depth guide on best practices for teleporting assets using XCM
+
+    [:octicons-arrow-right-24: Best Practices for Teleporting Assets](/develop/interoperability/best-practices-for-teleporting-assets/){target=\_blank}
```

Comment:
Since we’re at the end of the page and the goal of these cards is to guide users to the next step, we shouldn’t use target blank for internal links. For external links, we should use them so we can keep users on our docs site.